### DISCUSSION TITLE
Filename truncation preserve extension

### DISCUSSION THREAD

**Discussion by FluffyJay1** [leo-arch/clifm#150] [Ideas]
Hi,
Normally a filename like `thisisafile.txt` would get truncated to something like `thisisafi~`.
However I propose that we have an option that instead truncates files to something like `thisis~txt`.
This way you can still tell what type these files are at a glance, but the tilde in place of the period alerts you that it was truncated.
Yea/nay?

**Comment by leo-arch** [comment]
Hi @FluffyJay1. Sounds reasonable. That's what ranger does, isn't it?

I'll take a look at it.

**Comment by leo-arch** [comment]
Hey @FluffyJay1, extensions should be preserved now in trimmed file names.

Please take a look at it and report whatever issue comes up.

**Comment by FluffyJay1** [comment]
Wow that was quick. So far it's working perfectly. Thanks!
